worker/fanconfigurer/manifold.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2017 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package fanconfigurer
+
+import (
+	"github.com/juju/errors"
+	worker "gopkg.in/juju/worker.v1"
+
+	apiagent "github.com/juju/juju/api/agent"
+	"github.com/juju/juju/api/base"
+	"github.com/juju/juju/worker/dependency"
+)
+
+// ManifoldConfig defines the names of the manifolds on which a
+// Manifold will depend.
+type ManifoldConfig struct {
+	// These are the dependency resource names.
+	APICallerName string
+}
+
+// Manifold returns a dependency manifold that runs a log forwarding
+// worker, using the resource names defined in the supplied config.
+func Manifold(config ManifoldConfig) dependency.Manifold {
+	return dependency.Manifold{
+		Inputs: []string{
+			config.APICallerName,
+		},
+		Output: func(in worker.Worker, out interface{}) error {
+			inWorker, _ := in.(*FanConfigurer)
+			if inWorker == nil {
+				return errors.Errorf("in should be a %T; got %T", inWorker, in)
+			}
+			switch outPointer := out.(type) {
+			case *bool:
+				*outPointer = true
```

Comment:
It isn't ready until the config has been executed right?

As this stands, the machiner can start as soon as the fan configurer has started.

Reply:
The first iteration of configuration is done in the start function.